flashcards/web/src/modules/AddDeck/components/AddDecksForm.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from "react";
+import { Formik, Form, ErrorMessage } from "formik";
+import * as Yup from "yup";
+import { TextField, Typography, Button, Select, InputLabel, MenuItem, Checkbox, FormControlLabel } from "@mui/material";
+
+const validationSchema = Yup.object({
+    tagName: Yup.string().min(2, "Too Short!").max(20, "Too Long!").required("Required!"),
+    deckName: Yup.string().min(2, "Too Short!").max(20, "Too Long!").required("Required!"),
+    selectTag: Yup.string().required("Required!"),
+    selectCategory: Yup.string().required("Required!"),
+    checkboxPublic: Yup.boolean(),
+    checkboxForSale: Yup.boolean(),
+    price: Yup.number().when("checkboxForSale", {
+        is: true,
+        then: Yup.number().required("Required!"),
+    }),
+});
+
+const initialValues = {
+    tagName: "",
+    deckName: "",
+    selectTag: "",
+    selectCategory: "",
+    checkboxPublic: false,
+    checkboxForSale: false,
+    price: "",
+};
+
+function AddDecksForm() {
+    const [state, setState] = useState(initialValues);
+
+    const handleCheckboxPublicChange = (event) => {
+        setState({ ...state, checkboxPublic: event.target.checked });
+    };
+
+    const handleCheckboxForSaleChange = (event) => {
+        setState({ ...state, checkboxForSale: event.target.checked, price: event.target.checked });
+    };
+
+    const handleSubmit = (values, actions) => {
+        setTimeout(() => {
+            alert(JSON.stringify(values, null, 2));
+            actions.setSubmitting(false);
+        }, 1000);
+    };
+
+    return (
+        <div>
+            <Formik initialValues={initialValues} validationSchema={validationSchema} onSubmit={handleSubmit}>
+                {({ isSubmitting }) => (
+                    <div className="vh-100 flex flex-col justify-center items-center">
+                        <Form className="w-1/2">
```

Comment:
would consider sizing it differently, in case of ultra wide screens it's gonna be rather wide.
I'd probably add max witdth or breakpoints for width.